WooCommerce/UITestsFoundation/XCTest+Extensions.swift

```diff
@@ -119,10 +119,10 @@ extension XCUIElement {
         return XCUIApplication().staticTexts.containing(predicate).count
     }
 
-    public func assertTextVisibilityCount(textToFind: String, expectedCount: Int) {
-        let count = try! getStaticTextVisibilityCount(textToFind: textToFind)
-        XCTAssertTrue(count == expectedCount, "Expected '\(textToFind)' to appear \(expectedCount) times, but it appeared '\(count)' times!")
-    }
+    public func assertTextVisibilityCount(textToFind: String, expectedCount: Int = 1) {
```

Comment:
👍 for adding the default value!

Reply:
I'd say that using a default value in this particular case does not improve the explicitness of code. A person that sees this the first time:

```swift
app.assertTextVisibilityCount(textToFind: String(orders[0].id))
```

Will most likely want to take a look into the function to make sure that expected count is `1` for the default case. I'd suggest either adding an extra method (like `assertTextShownOnce` or a better name) which will call `assertTextVisibilityCount` in its turn, and use the new method for the cases when the expected number is one. Or, always provide the expected count value.

Another suggestion is that you could utilize a possibility to omit parameter names, and also have the `external` and `internal` parameter names, and adjust the naming a bit (because the name like `assertTextVisibilityCount ` already implies that one of the parameter will be text, and the other will be count):

```suggestion
    public func assertVisibilityCount(forText text: String, _ expectedCount: Int = 1) {
```

Which will allow you to have a call looking like this:

`app.assertVisibilityCount(forText: order.status, 1)`

(naming are just the examples)

Reply:
Just realized that this is an old method... So my suggestions from above are completely optional.

Reply:
I agree that adding the default value doesn't improve the explicitness/readability, however I needed a way to use `expectedCount: 2` since I'm calling this in SingleOrderScreen in order to check the first name, which appears twice in this case. Since most of the rest of the calls to this function do expect only 1 count, I made that the default.
Then, in SingleOrderScreen, I could assert:
`app.assertTextVisibilityCount(textToFind: order.billing.first_name, expectedCount: 2)`